token/services/interop/htlc/db/sql/tokendbext.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package sql
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"runtime/debug"
+
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/db/driver"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/interop/htlc"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/logging"
+	"github.com/pkg/errors"
+)
+
+var logger = logging.MustGetLogger("token-sdk.sql.htlc")
+
+type htlcTokenDBExtensionTables struct {
+	Tokens string
+}
+
+type HTLCTokenDBExtension struct {
+	table htlcTokenDBExtensionTables
+}
+
+func NewHTLCTokenDBExtension(table htlcTokenDBExtensionTables) *HTLCTokenDBExtension {
+	return &HTLCTokenDBExtension{table: table}
+}
+
+func (e *HTLCTokenDBExtension) GetSchema() string {
+	return fmt.Sprintf(`
+		-- Tokens
+		CREATE TABLE IF NOT EXISTS %s (
+			tx_id TEXT NOT NULL,
+			idx INT NOT NULL,
+			sender_raw BYTEA NOT NULL,
+			recipient_raw BYTEA NOT NULL,
+			deadline TIMESTAMP NOT NULL,
```

Comment:
All of these fields are already very nice to have! Now after this, it would be amazing to have the following two fields as well:

- preimage (easy; after scanning the blockchain it can be just updated in the table)
- status (trickier; depends on heuristics of whether the deadline is past and the preimage is available...)

I'm not sure how the process of these should work, like when they would be updated and if the 'interop' package already has the right hooks to manage that. But let's consider it in the design; if there is a way, it would help with the transparency and integrity of htlc transactions.

In terms of use cases, I'll share some diagrams and code with you that I used to approach this state on the application level.